api/scripts/dev/check-api-dependency-violations.js

```diff
@@ -0,0 +1,223 @@
+/* eslint-disable no-sync */
+import fs from 'node:fs';
+import { builtinModules } from 'node:module';
+import path from 'node:path';
+
+import { init, parse } from 'es-module-lexer';
+import { glob } from 'glob';
+
+import { Script } from '../../src/shared/application/scripts/script.js';
+import { ScriptRunner } from '../../src/shared/application/scripts/script-runner.js';
+
+await init;
+
+export class CheckApiDependencyViolations extends Script {
+  constructor() {
+    super({
+      description: 'Check API dependency violations between contexts [dev only]',
+      permanent: false,
+      options: {
+        report: {
+          type: 'string',
+          choices: ['json', 'md'],
+          describe: 'Report output file type',
+          requiresArg: true,
+          default: 'json',
+        },
+        context: {
+          type: 'string',
+          describe: 'Filter with an API context name',
+          requiresArg: true,
+        },
+      },
+    });
+  }
+
+  async handle({ options, logger }) {
+    config.basePath = path.resolve(import.meta.dirname, '../..');
+    if (!config.basePath.endsWith('/api')) {
+      throw new Error('Pix API folder not found.');
+    }
+
+    const excludedModules = [...getNodeModules(config), ...getBuiltinModules()];
+    const excludedPath = [...getExcludedPath(config), ...getExcludedApisPath(config)];
+
+    const violationsByEntry = new Map();
+    for (const entryPath of config.entries) {
+      if (options.context && !entryPath.startsWith(`src/${options.context}`)) {
+        continue;
+      }
+
+      const violationsByFile = await checkDependencyViolations(
+        config.basePath,
+        entryPath,
+        excludedModules,
+        excludedPath,
+      );
+      violationsByEntry.set(entryPath, violationsByFile);
+    }
+
+    const stats = computeGlobalStats(violationsByEntry);
+
+    let content = null;
+    if (options.report === 'json') {
+      content = JSON.stringify({ stats, violations: Object.fromEntries(violationsByEntry) });
+    } else if (options.report === 'md') {
+      content = reportToMarkdown(stats, violationsByEntry);
+    }
+
+    if (content) {
+      const reportFile = `${config.basePath}/dependencies-violations.${options.report}`;
+      fs.writeFileSync(reportFile, content);
+      logger.info(`Report file: ${reportFile}`);
```

Comment:
Suggestion : Préciser que c'est de la génération de fichier 😄 
```suggestion
      logger.info(`Generating report file: ${reportFile}`); 
```